Lib/test/test_type_aliases.py

```diff
@@ -0,0 +1,155 @@
+import types
+import unittest
+from test.support import check_syntax_error
+
+from typing import Callable, TypeAliasType, get_args, get_origin
+
+from .test_type_params import run_code
+
+
+class TypeParamsInvalidTest(unittest.TestCase):
+    def test_name_collision_01(self):
+        check_syntax_error(self, """type TA1[A, **A] = None""", "duplicate type parameter 'A'")
+
+    def test_name_non_collision_02(self):
+        run_code("""type TA1[A] = lambda A: None""")
+
+    def test_name_non_collision_03(self):
+        run_code("""\
+            class Outer[A]:
+                type TA1[A] = None
+            """
+        )
+
+
+class TypeParamsAccessTest(unittest.TestCase):
+    def test_alias_access_01(self):
+        ns = run_code("type TA1[A, B] = dict[A, B]")
+        alias = ns["TA1"]
+        self.assertIsInstance(alias, TypeAliasType)
+        self.assertEqual(alias.__type_params__, get_args(alias.__value__))
+
+    def test_alias_access_02(self):
+        run_code("""\
+            type TA1[A, B] = TA1[A, B] | int
+            """
+        )
+
+    def test_alias_access_03(self):
+        run_code("""\
+            class Outer[A]:
+                def inner[B](self):
+                    type TA1[C] = TA1[A, B] | int
+            """
+        )
+
+
+class TypeParamsAliasValueTest(unittest.TestCase):
+    def test_alias_value_01(self):
+        type TA1 = int
+
+        self.assertIsInstance(TA1, TypeAliasType)
+        self.assertEqual(TA1.__value__, int)
+        self.assertEqual(TA1.__parameters__, ())
+        self.assertEqual(TA1.__type_params__, ())
+
+        type TA2 = TA1 | str
+
+        self.assertIsInstance(TA2, TypeAliasType)
+        a, b = TA2.__value__.__args__
+        self.assertEqual(a, TA1)
+        self.assertEqual(b, str)
+        self.assertEqual(TA2.__parameters__, ())
+        self.assertEqual(TA2.__type_params__, ())
+
+    def test_alias_value_02(self):
+        class Parent[A]:
+            type TA1[B] = dict[A, B]
+
+        self.assertIsInstance(Parent.TA1, TypeAliasType)
+        self.assertEqual(len(Parent.TA1.__parameters__), 1)
+        self.assertEqual(len(Parent.__parameters__), 1)
+        a, = Parent.__parameters__
+        b, = Parent.TA1.__parameters__
+        self.assertEqual(Parent.__parameters__, (a,))
+        self.assertEqual(Parent.TA1.__type_params__, (b,))
+        self.assertEqual(Parent.TA1.__value__, dict[a, b])
+
+    def test_alias_value_03(self):
+        def outer[A]():
+            type TA1[B] = dict[A, B]
+            return TA1
+
+        o = outer()
+        self.assertIsInstance(o, TypeAliasType)
+        self.assertEqual(len(o.__parameters__), 1)
+        self.assertEqual(len(outer.__type_params__), 1)
+        b = o.__parameters__[0]
+        self.assertEqual(o.__type_params__, (b,))
+
+        def more_generic[T, *Ts, **P]():
```

Comment:
This looks like a separate test from the above.
```suggestion
    def test_alias_value_04(self):
        def more_generic[T, *Ts, **P]():
```